gutils/fsys.c

```diff
@@ -1018,6 +1018,83 @@ static int mime_comp(const void *k, const void *v) {
     return strmatch((const char*)k, ((const char**)v)[0]);
 }
 
+FILE *fmemoryopen(void* buffer, size_t size, const char* mode){
+    FILE *file = NULL;
+    #ifdef __unix__
+    file = fmemopen(buffer, size, mode);
+    #else
+    file = tmpfile();
+    if ( file==NULL ) {
+        fprintf(stderr, "Failed to create temporary file");
+        return NULL;
+    }
+    fwrite(buffer, size, 1, file);
```

Comment:
`fwrite()` puts the file-position indicator at the end of the buffer, which is probably undesirable. You would need to call `rewind()` or `fseek()` after it.